src/sensai/featuregen.py

```diff
@@ -269,6 +278,17 @@ def _generate(self, df: pd.DataFrame, ctx=None) -> pd.DataFrame:
         return resultDf
 
 
+def getFlattenedFeatureGenerator(fgen: FeatureGenerator, ruleTemplate: data_transformation.DFTNormalisation.RuleTemplate):
```

Comment:
This function is too specific. 
Why can I only apply it to flatten all columns?
Why can I only use a rule template?

Also, the prefix "get" in its name is not appropriate here.

Reply:
I extended the function. This extension necessitated a new feature in TakeColumns generator which I have implemented in the same commit